lib/tests/test_git.rs

```diff
@@ -3954,3 +3957,78 @@ fn test_shallow_commits_lack_parents() {
     // FIXME: new ancestors should be indexed
     assert!(!repo.index().has_id(&jj_id(&a)));
 }
+
+#[test]
+fn test_remote_remove_refs() {
+    let test_repo = TestRepo::init_with_backend(TestRepoBackend::Git);
+    let repo = &test_repo.repo;
+
+    add_remote(repo.store(), "foo", "https://example.com/").unwrap();
+    // Reload after Git configuration change.
+    let repo = &test_repo
+        .env
+        .load_repo_at_head(&testutils::user_settings(), test_repo.repo_path());
+
+    let git_repo = get_git_repo(repo);
+    empty_git_commit(&git_repo, "refs/remotes/foo/a", &[]);
+    empty_git_commit(&git_repo, "refs/remotes/foo/b", &[]);
+
```

Comment:
I think we should add more refs to test edge case behavior. e.g. `refs/remotes/foobar/a`, `refs/remotes/foo/x/y`, etc.

(also applies to rename test)

Reply:
Done, and split the commit adding the tests so that we can see what changes in the `.git/config` output too.